src/mlpack/core/util/arma_traits.hpp

```diff
@@ -301,15 +311,23 @@ struct IsBaseMatType<arma::SpRow<eT>>
 template<typename T>
 struct IsArma
 {
-  constexpr static bool value = arma::is_arma_type<T>::value;
+  constexpr static bool value
+  =  arma::is_arma_type<T>::value
+  || arma::is_arma_cube_type<T>::value
+  || arma::is_arma_sparse_type<T>::value
+  ;
 };
 
 #if defined(MLPACK_HAS_COOT)
 
 template<typename T>
 struct IsCoot
 {
-  constexpr static bool value = coot::is_coot_type<T>::value;
+  constexpr static bool value
+  =  coot::is_coot_type<T>::value
+  || coot::is_coot_cube_type<T>::value
+  //|| coot::is_coot_sparse_type<T>::value
+  ;
```

Comment:
```suggestion
  constexpr static bool value = coot::is_coot_type<T>::value ||
                                coot::is_coot_cube_type<T>::value;
```

Same as above (and there's no sparse type in Bandicoot).